### ISSUE TITLE
Dropout layers

### ISSUE THREAD

**Issue by GabbySuwichaya** [rusty1s/pytorch_geometric#2362]
## ❓ Questions & Help

Hi ! I have tried to extend some examples of your work for my own application such as [SuperGAT](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/super_gat.py) and [GAT](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/gat.py) as well as the associated message passing.  

Here, I have noticed that you have used the `dropout layers` on many occasions. So, I wonder what is the reason behind using the dropout layer? What are the benefits?

**Comment by rusty1s** [comment]
`Dropout` is a standard technique in DL, and in particular very useful in GNNs to avoid overfitting. In particular, the `GAT` paper mentions applying dropout to attention weights, leading to a dropout of edges during training. With this, the model has to learn to fit to a given task, even in the absence of edges or node features, improving generalization in return.

**Comment by GabbySuwichaya** [comment]
I see. Actually, I have also noticed that when I used `Dropout` ....The value of the attention becomes much smaller.... I am currently trying to normalize the features so that the value is around 0-1.... However, I am not sure what exactly I should do....  How do you normally handle this? ( I am so sorry I am very new to this area).

**Comment by rusty1s** [comment]
The attention coefficients should be already normalized via `softmax`. If you want to normalize your input features, you can do something like
```python
x = x / x.norm(dim=-1, keepdim=True)
```

**Comment by GabbySuwichaya** [comment]
I see. Thanks so much. I will try.
